File: dangerfile.js

import { danger } from "danger";


// CAMBIOS EN ARCHIVO SWIFT  ***************************
const docsSwift = danger.git.fileMatch("**/*.swift")
if (docsSwift.edited) {
 message("MESSAGE !!!Se han realizado cambios en archivo .swift")
 } 

//CAMBIOS DE VERSION  ***************************
const versionCheck = danger.git.fileMatch("**/*.pbxproj")
if (versionCheck.edited) {
  warn("WARNING !!!Se han realizado cambios de Version del proyecto")
}

// CAMBIOS BSMOPERATIONFACTORY
const operationFactory = danger.git.fileMatch("**/*BSMOperationFactory")
if (operationFactory.edited) {
  fail("Se han realizado cambios en el archivo BSMOperationFactory")
 } 

// CAMBIOS BSMOPERATIONFACTORY
const viewChek = danger.git.fileMatch("**/*ViewController")
if (viewChek.edited) {
  fail("Se han realizado cambios en el archivo ViewController")
 } 

//CAMBIOS DE LIBRERIA  ***************************
const packageChanged = danger.git.modified_files.includes('Carthage');
if (packageChanged.edited) {
   fail(" FAIL !!! Se han realizado cambios a nivel de .Carthage")
}

File: Dangerfile.swift
import Danger 
let danger = Danger()


// TEST MESSAGE FROM DANGER ***************************
danger.message("Hello from Danger GITHUB Prueba DangerFile")

// CAMBIOS DE VERSION  ***************************
if danger.git.modifiedFiles.contains(where: { $0.contains(".xcworkspacedata")}) {
        warn(" WARNING !!! Se han realizado cambios a nivel de .xcproject")
}
// CAMBIOS DE LIBRERIA  ***************************
if danger.git.modifiedFiles.contains(where: { $0.contains("Pods")}) {
        warn(" WARNING !!! CAMBIO EN LIBRERIAS DE APP ")
    }


if danger.git.modifiedFiles.contains(where: { $0.contains(".swift")}) {
        warn(" WARNING !!! Se han realizado cambios a nivel de SWIFT")
}
